### DISCUSSION TITLE
Z-loss definition

### DISCUSSION THREAD

**Discussion by Jbiloki** [google-research/t5x#1577] [Q&A]
Hey folks, I was looking into the loss function [here](https://github.com/google-research/t5x/blob/efce74c982db9937117ae37a873d2e4c67ab4d56/t5x/losses.py#L55) and see there is a z loss added to the typical log loss definition.

The only paper I could surface on this was https://arxiv.org/abs/1604.08859 but I don't see the correlation. In the code it is logsumexp of the logits squared.

But in the paper it is calculating mean/variance for the z-loss you can see the author's implementation [here](https://github.com/pascal20100/factored_output_layer/blob/master/example/timing_example.py#L57) I am asking for help either connecting the two or looking for resources on the current implementation if they are infact unrelated.

Thank you.
